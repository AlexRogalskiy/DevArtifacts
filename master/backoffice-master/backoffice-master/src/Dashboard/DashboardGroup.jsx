import React from 'react'
import PropTypes from 'prop-types'

import {
  Grid,
  Typography,
  withStyles,
} from '@material-ui/core'

import DashboardCard from './DashboardCard'

const styles = theme => ({
  headline: {
    marginTop: theme.spacing.unit * 3,
    marginBottom: theme.spacing.unit * 2,
  },
})

const DashboardGroup = ({
  title,
  cards,
  onClick,
  classes,
}) => (
  <div>
    <div className={classes.headline}>
      <Typography variant="headline">
        {title}
      </Typography>
    </div>

    <Grid container spacing={16}>

      {cards ? cards.map(card => (
        <Grid item xs={12} sm={4} key={`group-card-${card.id}`}>
          <DashboardCard
            title={card.title}
            description={card.description}
            handleClick={() => onClick(card.link)}
            icon={card.icon}
            isDisabled={card.isDisabled}
          />
        </Grid>
      )) : null}

    </Grid>
  </div>
)

DashboardGroup.propTypes = {
  title: PropTypes.string.isRequired,
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(DashboardGroup)
