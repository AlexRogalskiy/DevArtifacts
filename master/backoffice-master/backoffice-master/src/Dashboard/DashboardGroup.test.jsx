import React from 'react'
import { mount } from 'enzyme'

import DashboardGroup from './DashboardGroup'

it('renders correctly', () => {
  const tree = mount(<DashboardGroup
    title="Title"
    cards={[{
      title: 'title',
      description: 'desc',
    }]}
    onClick={() => {}}
  />)

  expect(tree).toMatchSnapshot()
})
