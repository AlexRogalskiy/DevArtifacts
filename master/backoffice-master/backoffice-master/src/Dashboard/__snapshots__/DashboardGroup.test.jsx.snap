// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="DashboardGroup-headline-1"
  >
    <h1
      className="MuiTypography-root-2 MuiTypography-headline-7"
    >
      Title
    </h1>
  </div>
  <div
    className="MuiGrid-container-27 MuiGrid-spacing-xs-16-50"
  >
    <div
      className="MuiGrid-item-28 MuiGrid-grid-xs-12-67 MuiGrid-grid-sm-4-73"
    >
      <div
        className="MuiPaper-root-131 MuiPaper-elevation1-134 MuiPaper-rounded-132 MuiCard-root-130 DashboardCard-root-124"
        onClick={[Function]}
      >
        <div
          className="MuiCardContent-root-158 DashboardCard-content-125"
        >
          <h2
            className="MuiTypography-root-2 MuiTypography-headline-7 DashboardCard-title-127"
          >
            title
          </h2>
          <p
            className="MuiTypography-root-2 MuiTypography-body1-11"
          >
            desc
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
